src/Startup.cpp

```diff
@@ -111,20 +123,35 @@ grid->addWidget(new WText("GIS Data Directory    "),0,1);
 
 Wt::WSelectionBox *datadir = new Wt::WSelectionBox();
 datadir->resize(200,25);
-datadir->addItem("/home/mayank/webgrassdata");   
+datadir->addItem(m_GrassDataDirectory);   
 grid->addWidget(datadir,0,2);
 
 
 
 
-const fs::path fi= "/home/mayank/webgrassdata";
+const fs::path fi= m_GrassDataDirectory;
 vector<string> dir;
 show_files(fi,dir,true);
 Wt::WSelectionBox *cb = new Wt::WSelectionBox();
 cb->resize(200,240);
-cb->addItem(dir[0]);
-cb->addItem(dir[1]);
-cb->addItem(dir[2]);
+
+
+for(vector<string>::iterator it = dir.begin();it!=dir.end();++it)
+{
+	bool isValidLocation = checkExistance(*it, m_GrassDataDirectory);
+	cout<<isValidLocation<<endl;
+	if (isValidLocation) 
+		{ cb->addItem(*it); } 
+}
+ 
+
+
+// bool isValidLocation = checkExistance(dir[0], m_GrassDataDirectory);
+// if (isValidLocation) 
+// 	{ cb->addItem(dir[0]); } 
+// cb->addItem(dir[1]);
+// cb->addItem(dir[2]);
+
 cb->setCurrentIndex(1); 
```

Comment:
lines 132-155 could be a simple function because it is repeated for cb and cb1. 

updateComboBox(WtComboBoxLocation *loc)
in that function pass 'cb' variable.
updateComboBox(cb)

updateComboBox(cb2)